won't start after last update

Hi just updated this morning but got a problem, docker won't restart.

was working great before and v useful, thanks for it.

any ideas?

```
Loading settings from /config
fs.js:663
  return binding.open(pathModule.toNamespacedPath(path),
                 ^

Error: ENOENT: no such file or directory, open '/config/configuration.yaml'
    at Object.fs.openSync (fs.js:663:18)
    at Object.fs.readFileSync (fs.js:568:33)
    at loadFile (/usr/src/app/node_modules/convict/lib/convict.js:454:19)
    at /usr/src/app/node_modules/convict/lib/convict.js:613:24
    at Array.forEach (<anonymous>)
    at Object.loadFile (/usr/src/app/node_modules/convict/lib/convict.js:611:13)
    at Object.<anonymous> (/usr/src/app/default_settings.js:55:8)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ha-dockermon@0.0.13 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ha-dockermon@0.0.13 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-09-06T06_21_57_082Z-debug.log

```

Hey Phil! Same issue here. Recently update won't start container. 
Already tried chmoding the config dir/file 0777, no go. 

```
Loading settings from /config
Warning: configuration param 'docker_connection.port' not declared in the schema
/usr/src/app/node_modules/convict/lib/convict.js:679
          throw new Error(output);
          ^

Error: docker_connection.host: must be of type String
configuration param 'docker_connection.port.doc' missing from config, did you override its parent?
configuration param 'docker_connection.port.format' missing from config, did you override its parent?
    at Object.validate (/usr/src/app/node_modules/convict/lib/convict.js:679:17)
    at Object.<anonymous> (/usr/src/app/default_settings.js:58:8)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/index.js:6:14)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ha-dockermon@0.0.13 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ha-dockermon@0.0.13 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-09-06T16_30_49_632Z-debug.log
```

> Hey Phil! Same issue here. Recently update won't start container.
> Already tried chmoding the config dir/file 0777, no go.
> 
> ```
> > ha-dockermon@0.0.13 start /usr/src/app
> > node index.js
> 
> Loading settings from /config
> Warning: configuration param 'docker_connection.port' not declared in the schema
> /usr/src/app/node_modules/convict/lib/convict.js:679
>           throw new Error(output);
>           ^
> 
> Error: docker_connection.host: must be of type String
> configuration param 'docker_connection.port.doc' missing from config, did you override its parent?
> configuration param 'docker_connection.port.format' missing from config, did you override its parent?
>     at Object.validate (/usr/src/app/node_modules/convict/lib/convict.js:679:17)
>     at Object.<anonymous> (/usr/src/app/default_settings.js:58:8)
>     at Module._compile (module.js:660:30)
>     at Object.Module._extensions..js (module.js:671:10)
>     at Module.load (module.js:573:32)
>     at tryModuleLoad (module.js:513:12)
>     at Function.Module._load (module.js:505:3)
>     at Module.require (module.js:604:17)
>     at require (internal/module.js:11:18)
>     at Object.<anonymous> (/usr/src/app/index.js:6:14)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! ha-dockermon@0.0.13 start: `node index.js`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the ha-dockermon@0.0.13 start script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2019-09-06T16_30_49_632Z-debug.log
> ```
> 
> My configuration.yaml is similar to the following:
> 
> ```
> debug: false
> http:
>   port: 8126
>   username: foo
>   password: bar
> docker_connection:
>   type: socket
>   path: /var/run/docker.sock
>   host:
>   port:
> ```

I came here due to the same original issue.  From this thread I was able to create a configuration.yaml that worked. Just remove the host and port lines in your configuration.yaml and it should be fine.   Unfortunately mine starts but then I get 401 errors each time I query it and nothing in the log to explain.

--- a/config/configuration.yaml.sample
+++ b/config/configuration.yaml.sample
@@ -1,10 +1,6 @@
 debug: false
 http:
   port: 8126
-  username: 
-  password:
 docker_connection:
   type: socket
   path: /var/run/docker.sock
-  host:
-  port:

--- a/default_settings.js
+++ b/default_settings.js
@@ -1,5 +1,6 @@
 var convict = require('convict');
     yaml = require('js-yaml');
+    fs = require('fs');
 
 var config = convict({
   debug: {
@@ -16,12 +17,12 @@ var config = convict({
       username: {
           doc: 'Optional. The HTTP Username for authentication',
           format: String,
-          default: ''
+          default: undefined
       }, 
       password: {
         doc: 'Optional. The HTTP Password for authentication',
         format: String,
-        default: '',
+        default: undefined,
         sensitive: true
     }
   },
@@ -37,13 +38,14 @@ var config = convict({
       default: '/var/run/docker.sock'
     },
     host: {
-      doc: 'URL to remote docker host',
+      doc: 'Optional. URL to remote docker host',
       format: String,
-      default: ''
+      default: undefined
     },
     port: {
-      doc: 'Port on remote docker host',
-      format: 'port_or_windows_named_pipe'
+      doc: 'Optional. Port on remote docker host',
+      format: 'port_or_windows_named_pipe',
+      default: undefined
     }
   }
 });
@@ -52,7 +54,15 @@ var config = convict({
 convict.addParser({ extension: ['yml', 'yaml'], parse: yaml.safeLoad });
 var config_dir = process.env.config_dir || "./config";
 console.log("Loading settings from " + config_dir);
-config.loadFile(config_dir + '/configuration.yaml');
+
+//Check if the configuration file exists, if it doesn't then skip this
+try{
+  if (fs.existsSync(config_dir + '/configuration.yaml')) {
+    config.loadFile(config_dir + '/configuration.yaml');
+  }
+} catch(err) {
+  console.warn("No configuration file detected. Using default settings");
+}
 
 // Perform validation
 config.validate({allowed: 'warn'});

--- a/readme.md
+++ b/readme.md
@@ -30,8 +30,8 @@ You can change some configuration options for this service by editing config/con
 | http.password          | If set all calls to this service must use HTTP basic authentication with this password. **If set, must not be an empty value.**                         | None                 |
 | docker_connection.type | How the service connects to docker. Valid options are socket or http                                             | socket               |
 | docker_connection.path | The path to the Docker socket. Useful when running this service on the host directly (not in a Docker container) | /var/run/docker.sock |
-| docker_connection.host | The host IP/Domain of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`. If `docker_connection.type` is set to `http` do not add this to your configuration YAML.         | None                 |
-| docker_connection.port | The port of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`.  If `docker_connection.type` is set to `http` do not add this to your configuration YAML.                  | None                 |
+| docker_connection.host | The host IP/Domain of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`. **If set, must not be an empty value.**         | None                 |
+| docker_connection.port | The port of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`.  **If set, must not be an empty value.**                  | None                 |
 
 ### Connecting to Docker
 

--- a/readme.md
+++ b/readme.md
@@ -25,13 +25,13 @@ You can change some configuration options for this service by editing config/con
 | Option                 | Description                                                                                                      | Default Value        |
 |------------------------|------------------------------------------------------------------------------------------------------------------|----------------------|
 | debug                  | If set to true, will output some helpful debug messages to the console.                                          | false                |
-| http.port              | The HTTP port the service listens on.                                                                            | 8126                 |
-| http.username          | If set all calls to this service must use HTTP basic authentication with this username.                          | None                 |
-| http.password          | If set all calls to this service must use HTTP basic authentication with this password.                          | None                 |
+| http.port              | The HTTP port the service listens on. Must be a valid HTTP port or windows socket.                                                                          | 8126                 |
+| http.username          | If set all calls to this service must use HTTP basic authentication with this username. **If set, must not be an empty value.**                         | None                 |
+| http.password          | If set all calls to this service must use HTTP basic authentication with this password. **If set, must not be an empty value.**                         | None                 |
 | docker_connection.type | How the service connects to docker. Valid options are socket or http                                             | socket               |
 | docker_connection.path | The path to the Docker socket. Useful when running this service on the host directly (not in a Docker container) | /var/run/docker.sock |
-| docker_connection.host | The host IP/Domain of the host Docker to connect to. Only used when docker_connection.type is set to http        | None                 |
-| docker_connection.port | The port of the host Docker to connect to. Only used when docker_connection.type is set to http                  | None                 |
+| docker_connection.host | The host IP/Domain of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`. If `docker_connection.type` is set to `http` do not add this to your configuration YAML.         | None                 |
+| docker_connection.port | The port of the host Docker to connect to. Only used when `docker_connection.type` is set to `http`.  If `docker_connection.type` is set to `http` do not add this to your configuration YAML.                  | None                 |
 
 ### Connecting to Docker
 

